src/platform/plugins/shared/vis_types/timeseries/public/convert_to_lens/lib/convert/terms.ts

```diff
@@ -45,7 +47,7 @@ const getOrderByWithAgg = (series: TermsSeries, columns: Column[]): OrderByWithA
     return false;
   });
 
-  if (!orderAgg) {
+  if (!orderAgg || !isColumnMetricWithSourceField(orderAgg)) {
```

Comment:
This typeguard guarantees that only metric charts would be assigned as ranking criteria for a terms column.
The problem was hidden with the previous loose types, but it surfaced when adopting the original Lens types, so it required to be correctly handled.